src/Umbraco.Core/Services/ContentService.cs

```diff
@@ -2008,6 +2008,8 @@ private Attempt<PublishStatus> SaveAndPublishDo(IContent content, int userId = 0
                     content.WriterId = userId;
 
                     repository.AddOrUpdate(content);
+                    
+                    content = repository.Get(content.Id);

```

Comment:
@clausjensen is this the only way to solve it, by getting the content after Adding or Updating?
Should the `AddOrUpdate` method be updated to work instead perhaps?

Off to test this out in the meantime, but by looking at this change I assume it will 'just' work :smile: